lib/pages/caregiver/forms/plan_form_page.dart

```diff
@@ -55,35 +56,45 @@ class _CaregiverPlanFormPageState extends State<CaregiverPlanFormPage> {
 		formKey = GlobalKey<FormState>();
     return BlocConsumer<PlanFormCubit, PlanFormState>(
 			listener: (context, state) {
-				if (state is PlanFormSubmissionSuccess)
-					Navigator.of(context).pop(); // TODO also show some visual feedback?
-				else if (state is PlanFormDataLoadSuccess)
+				if (state is PlanFormSubmissionSuccess) {
+					if(formType == AppFormType.copy)
+						Navigator.of(context).pushNamed(AppPage.caregiverPlans.name);
```

Comment:
Not a good idea: what if user presses _back_? Either pop once or twice or, if you want to navigate to caregiver plans you can try: `Navigator.of(context).popUntil(ModalRoute.withName(AppPage.caregiverPanel.name));` before pushing plans. **Could be off by one, test first**